var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -394,9 +394,14 @@ def set_blas_lapack(self):
 
     @when("@1.26:")
     def setup_build_environment(self, env):
-        # https://github.com/scipy/scipy/issues/19357
         if self.spec.satisfies("%apple-clang@15:"):
+            # https://github.com/scipy/scipy/issues/19357
             env.append_flags("LDFLAGS", "-Wl,-ld_classic")
+        elif self.spec.satisfies("%msvc"):
+            # For meson build system, compiler paths must be in quotes
+            # to prevent paths from being split by spaces.
+            env.set("CC", f'"{self.compiler.cc}"')
+            env.set("CXX", f'"{self.compiler.cxx}"')
```

Comment:
Want to hear @rgommers' opinion on this

Reply:
If this works, I am fine with this - can't see how it hurts. 

I'd caution though that for MSVC, setting paths to compilers is normally not enough. It needs activation via a script like `vcvarsall.bat` shipped with MSVC. The way to auto-activate MSVC with `meson` is to pass the `--vsenv` flags to `meson setup`. That is a no-op on other platforms, and should always do the right thing on Windows when you want to use MSVC.

Reply:
I wonder if this is something that should be "fixed" in meson @eli-schwartz  @michaelkuhn

Reply:
`%msvc` should be calling the appropriate `vcvars` script (in `Msvc.setup_custom_environment`)

It seems like this would still be a problem whether that is called: `meson` would split `CC` either way

We always set CC in for builds (in `build_environment`) and I wonder if we should be putting it in quotes always for Windows (if we did, there would not be a need to set it specifically in `py-numpy`). If that doesn't break other builds.

Reply:
> I wonder if we should be putting it in quotes always for Windows (if we did, there would not be a need to set it specifically in py-numpy). If that doesn't break other builds.

My first fix for this (noted in the description of fixes above) was to globally put the paths in quotes in `msvc.py`, but that caused CMake configuration failures for other packages.